soweego/importer/discogs_dump_extractor.py

```diff
@@ -78,140 +81,179 @@ def extract_and_populate(self, dump_file_paths: Iterable[str], resolve: bool):
         LOGGER.info('SQL tables dropped and re-created: %s',
                     [table.__tablename__ for table in tables])
 
-        with gzip.open(dump_file_path, 'rt') as dump:
+        extracted_path = '.'.join(dump_file_path.split('.')[:-1])
 
-            # count number of lines and move again up to the beginning
-            # of the file
-            n_rows = sum(1 for line in dump)
-            dump.seek(0)
+        # Extract dump file if it has not yet been extracted
+        if not os.path.exists(extracted_path):
 
-            session = db_manager.new_session()
+            LOGGER.info('Extracting dump file')
 
-            # we count how many iterations we've done up until now
-            # so that we can do an insertion every `self._sqlalchemy_commit_every`
-            # loops
-            e_counter = 0
+            with gzip.open(dump_file_path, 'rb') as f_in:
+                with open(extracted_path, 'wb') as f_out:
+                    shutil.copyfileobj(f_in, f_out)
```

Comment:
I'm not sure why you are doing this. Could you please clarify?

Reply:
The reason is because the new library we use, `lxml`, expects the path to the `xml` file, and not the file itself. So here we're saving the extracted file to disc, and passing it's new path to the `lxml.parse` method below (in the `g_process_et_items` function)

Reply:
Sounds good, thanks for the explanation!